src/snowcli/cli/common/cli_global_context.py

```diff
@@ -0,0 +1,211 @@
+from typing import Optional
+
+from snowcli.output.formats import OutputFormat
+from snowcli.snow_connector import connect_to_snowflake
+from snowflake.connector import SnowflakeConnection
+
+
+class _ConnectionContext:
+    def __init__(self):
+        self._cached_connection: Optional[SnowflakeConnection] = None
+
+        self._connection_name: Optional[str] = None
+        self._account: Optional[str] = None
+        self._database: Optional[str] = None
+        self._role: Optional[str] = None
+        self._schema: Optional[str] = None
+        self._user: Optional[str] = None
+        self._password: Optional[str] = None
+        self._authenticator: Optional[str] = None
+        self._private_key_path: Optional[str] = None
+        self._warehouse: Optional[str] = None
+        self._temporary_connection: bool = False
+
+    def __setattr__(self, key, value):
+        super.__setattr__(self, key, value)
+        if key is not "_cached_connection":
+            self._cached_connection = None
```

Comment:
Why it's needed?

Reply:
To invalidate cache after connection parameters changes

Reply:
Let's add comment about it

Reply:
Done